invokeai/frontend/web/src/features/parameters/types/parameterSchemas.ts

```diff
@@ -141,6 +141,20 @@ export const isParameterStrength = (val: unknown): val is ParameterStrength =>
   zParameterStrength.safeParse(val).success;
 // #endregion
 
+// #region SeamlessX
+const zParameterSeamlessX = z.boolean();
+export type ParameterSeamlessX = z.infer<typeof zParameterSeamlessX>;
+export const isParameterSeamlessX = (val: unknown): val is boolean =>
+  zParameterSeamlessX.safeParse(val).success && val !== null && val !== undefined;
```

Comment:
- The extra null/undefined checks are unnecessary, zod will do them for ya - just need to check `.success`
- Let's use `val is ParameterSeamlessX` instead of `val is boolean`

PS: This file is really verbose. We could use a helper to make it a lot more concise. In case you feel inspired to clean it up, here's a draft of some helpers:
```ts
/**
 * Helper to create a type guard from a zod schema. The type guard will infer the schema's TS type.
 * @param schema The zod schema to create a type guard from.
 * @returns A type guard function for the schema.
 */
const buildTypeGuard = <T extends z.ZodTypeAny>(schema: T) => {
  return (val: unknown): val is z.infer<T> => schema.safeParse(val).success;
};

/**
 * Helper to create a zod schema and a type guard from it.
 * @param schema The zod schema to create a type guard from.
 * @returns A tuple containing the zod schema and the type guard function.
 */
const buildParameter = <T extends z.ZodTypeAny>(schema: T) => [schema, buildTypeGuard(schema)] as const;

// #region Some parameter
export const [zParameter, isParameter] = buildParameter(z.string());
export type Parameter = z.infer<typeof zParameter>;
// #endregion
```

Only catch, `knip` is gonna complain about some unused exports... could just exclude this file in `invokeai/frontend/web/knip.ts`?

Reply:
a) Made the requested changes. 

For reference, the implementation is based on this. Not quite sure why the boolean parameter has the additional checks.
```
// #region HRF Enabled
const zParameterHRFEnabled = z.boolean();
export type ParameterHRFEnabled = z.infer<typeof zParameterHRFEnabled>;
export const isParameterHRFEnabled = (val: unknown): val is boolean =>
  zParameterHRFEnabled.safeParse(val).success && val !== null && val !== undefined;
// #endregion
```

b) I could possibly have a look at it next week. 

It adds an extra line, but could be more selective with the export?
Or one big export at the end of the file?
```
// #region Some parameter
const [zParameter, isParameter] = buildParameter(z.string());
export { isParameter };
export type Parameter = z.infer<typeof zParameter>;
// #endregion
```

Reply:
Thanks! Dunno why there were extra checks in that spot.

If you wanted to do the cleanup, I think I'd prefer the exports to be close to the definitions.